### PR TITLE
[e2e] story_microshift ensures operational status for cluster 

### LINKED ISSUE #3851: [QE] Story microshift fails when running remotely
When we run `@story-microshift` locally on a target test machine (mac or windows) it works as expected but when we run through the crc-e2e image remotely it fails adding the url for the service 

Logs from routes-controller:

```
ec2-user@Mac-mini ~ % oc logs -n openshift-ingress routes-controller-c44f58c8d-tv7fr
W0928 13:43:54.811595       1 client_config.go:617] Neither --kubeconfig nor --master was specified.  Using the inClusterConfig.  This might not work.
time="2023-09-28T13:43:54Z" level=info msg="Added service 'kubernetes' of type 'ClusterIP'"
time="2023-09-28T13:43:54Z" level=info msg="Added service 'router-internal-default' of type 'ClusterIP'"
time="2023-09-28T13:43:54Z" level=info msg="Added service 'dns-default' of type 'ClusterIP'"
time="2023-09-28T13:43:56Z" level=info msg="Service 'dns-default' of type 'ClusterIP' changed!"
time="2023-09-28T13:44:03Z" level=info msg="Added service 'httpd-example' of type 'ClusterIP'"
time="2023-09-28T13:44:03Z" level=info msg="added: httpd-example httpd-example-testproj.apps.crc.testing"
time="2023-09-28T13:44:03Z" level=error msg="Post \"http://gateway/hosts/add\": dial tcp: lookup gateway on 10.43.0.10:53: read udp 10.42.0.6:35808->10.43.0.10:53: read: connection refused"
time="2023-09-28T13:48:54Z" level=info msg="Service 'kubernetes' of type 'ClusterIP' changed!"
time="2023-09-28T13:48:54Z" level=info msg="Service 'router-internal-default' of type 'ClusterIP' changed!"
time="2023-09-28T13:48:54Z" level=info msg="Service 'dns-default' of type 'ClusterIP' changed!"
time="2023-09-28T13:48:54Z" level=info msg="Service 'httpd-example' of type 'ClusterIP' changed!"

```

### PR DESCRIPTION
**Pull request by adrianriobo**
Fix #3851 

this commit may will be partially reverted on a following PR in case #3852 will be implemented

### CODE REVIEW

File: test/e2e/testsuite/testsuite.go [hunk 1]
```diff
@@ -991,3 +995,39 @@ func DecodeBase64File(inputFile, outputFile string) error {
...
+
+func EnsureUserNetworkmode() error {
+	if runtime.GOOS == "linux" {
+		return crcCmd.SetConfigPropertyToValueSucceedsOrFails(
+			"network-mode", "user", "succeeds")
+	}
+	return nil
+}
+
+// This function will wait until the microshift cluster got operational
```
Comment (jsliacan) [inline test/e2e/testsuite/testsuite.go:1007]: I have no comments, just maybe that the func doc comment usually starts with func name and describes output. But that's super minor.

**Review by jsliacan — APPROVED** [review]
LGTM

**Comment by praveenkumar** [comment]
@adrianriobo can we hold this a bit because I think some of those should be part of `crc start` so that when it start it should in working state.

**Comment by adrianriobo** [comment]
I would prefer to add this for the time being and then revert once you got the fix on `start` otherwise I can not test the Microshift scenario properly 

What do you think you can just add your commit on your PR and the revert commit for this in there as well?

**Pull request merged**